### PR TITLE
Channels & Frames

### PR DESCRIPTION
**Pull request by achaikou**
First 2 commits are the same as in other PR as it was easier to include them than not.

### CODE REVIEW

**Comment by jokva** [comment]
Isn't it fundamentally better to move this to methods on the frame itself, and then have `channel.curve()` use that?

File: python/dlisio/ext/core.cpp [hunk 1]
```diff
@@ -321,13 +321,15 @@ framesize frame_size(const char* fmt) {
...
 
-    return framesize { src, dst };
+    return fdatasize { src, dst };
 }
 
-void read_all_fdata(const char* fmt,
-                    dl::stream& file,
-                    const std::vector< int >& indices,
-                    py::buffer dstb)
+void read_fdata(const char* pre_fmt,
```
Comment (ErlendHaa) [inline python/dlisio/ext/core.cpp]: This is sweet, maybe it deserves its own commit ? (read_all_fdata -> read_fdata)

File: python/tests/test_frames.py [hunk 2]
```diff
@@ -75,14 +75,6 @@ def test_instance_dtype_fmt():
     assert 'fDDD' == frame.fmtstr()
     assert ('x-TIME 0~0', 'TDEP', 'x-TIME 1~0') == frame.dtype.names
 
-def test_instance_dtype_fmt():
```
[the remark this answers is not in the document]
Comment (achaikou) [reply python/tests/test_frames.py]: It's the same as the code above.

File: python/dlisio/__init__.py [hunk 3]
```diff
@@ -254,11 +254,10 @@ def load(self, sets=None):
...
         return self
 
-    def curves(self, fingerprint):
-        frame = self.indexedobjects['FRAME'][fingerprint]
-        fmt = frame.fmtstr()
-        indices = self.fdata_index[fingerprint]
-        a = np.empty(shape = len(indices), dtype = frame.dtype)
+    def curves(self, frame, dtype, fmt):
+        indices = self.fdata_index[frame.fingerprint]
+        #note: shape is wrong for multiple data in one frame
```
[the remark this answers is not in the document]
Comment (achaikou) [reply python/dlisio/__init__.py]: We already do in core where we detect it. This is just an indication note that once we support multiple data in one frame, this code should be affected as well.

File: python/tests/test_frames.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply python/tests/test_frames.py]: Oh right, didn't expand

File: python/dlisio/ext/core.cpp (hunk 1 above)
Comment (achaikou) [reply python/dlisio/ext/core.cpp]: To me that's the part of adding channel functionality. Change makes little sense without it, so I am not convinced. But we can discuss.

File: python/dlisio/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply python/dlisio/__init__.py]: Maybe remove dlis.curves completely? It feels a bit redundant when we have frames.curves and channels.curves
